hr_attendance_sheet/models/hr_attendance.py

```diff
@@ -0,0 +1,139 @@
+    auto_lunch = fields.Boolean(
+        string="Auto Lunch Applied",
+        help="If Auto Lunch is enabled and applied on this attendance.")
+    company_id = fields.Many2one(
+        'res.company',
+        string="Company",
+        related="attendance_sheet_id.company_id")
+
+    # Get Methods
+    def _get_attendance_employee_tz(self, date=None):
+        """ Convert date according to timezone of user """
+        tz = False
+        if self.employee_id:
+            tz = self.employee_id.tz
+        if not date:
+            return False
+        time_zone = pytz.timezone(tz or 'UTC')
+        attendance_dt = datetime.strptime(str(date),
+                                          DEFAULT_SERVER_DATETIME_FORMAT)
+        attendance_tz_dt = pytz.UTC.localize(attendance_dt)
+        attendance_tz_dt = attendance_tz_dt.astimezone(time_zone)
+        attendance_tz_date_str = datetime.strftime(
+            attendance_tz_dt, DEFAULT_SERVER_DATE_FORMAT)
+        return attendance_tz_date_str
+
+    def _get_attendance_state(self):
+        """ Check and raise error if related sheet is not in 'draft' state """
+        if self._context.get('allow_modify_confirmed_sheet', False):
+            return
+        if self.attendance_sheet_id and \
+                self.attendance_sheet_id.state == 'locked':
+            raise UserError(_(
+                "You cannot modify an entry in a locked sheet."
+            ))
+        elif self.attendance_sheet_id.state == 'done' and \
+                self.env.user not in \
+                self.attendance_sheet_id._get_possible_reviewers():
+            raise UserError(_(
+                "You cannot modify an entry in a approved sheet"
+            ))
+        return True
+
+    # Compute Methods
+    @api.depends('employee_id', 'check_in', 'check_out')
+    def _compute_attendance_sheet_id(self):
+        """ Find and set current sheet in current attendance record """
+        for attendance in self:
+            sheet_obj = self.env['hr.attendance.sheet']
+            check_in = False
+            if attendance.check_in:
+                check_in = attendance._get_attendance_employee_tz(
+                    date=attendance.check_in)
+                domain = [('employee_id', '=', attendance.employee_id.id)]
+                if check_in:
+                    domain += [
+                        ('date_start', '<=', check_in),
+                        ('date_end', '>=', check_in),
+                    ]
+                    attendance_sheet_ids = sheet_obj.search(domain, limit=1)
+                    attendance.attendance_sheet_id = \
+                        attendance_sheet_ids or False
+
+    @api.multi
+    def _compute_duration(self):
+        for rec in self:
+            if rec.check_in and rec.check_out:
+                start = fields.Datetime.from_string(rec.check_in)
+                end = fields.Datetime.from_string(rec.check_out)
+                delta = end - start
```

Comment:
```suggestion
                delta = rec.check_out - rec.check_in
```

rec.check_in and check_out is type datetime.
so no more fields.Datetime.from_string is needed